File: src/diaryApp/Diary.java
package diaryApp;

import javax.swing.*;
import java.util.ArrayList;

public class Diary {
    private Entry entry;
    private boolean isLocked = true;
    private String username;
    private String password;
    private ArrayList<Entry> entries = new ArrayList<>();


    public Diary(String username, String password) {
        this.username = username;
        this.password = password;
    }
    public String getUsername() {
        return username;
    }

    public boolean isLocked() {
        return isLocked;
    }

    public int getId(){
        return entries.size() + 1;
    }

    public void unlock(String password) {
        if(validatePassword(password)) {
            isLocked = false;
        }else{
            throw new IllegalArgumentException("Password does not match");
        }
    }

    public void lock(){
        isLocked = true;
    }

    public void createEntry(String title, String body) {
        if(!isLocked()){
            int  id = getId();
            Entry newEntry = new Entry(id, title, body);
            entries.add(newEntry);
            JOptionPane.showMessageDialog(null,"New entry created\nYour Entry Id :" + id);
        }else{
            throw new IllegalArgumentException("Diary is locked unlock first");
        }
    }

    public Entry findEntry(int id) {
        if (isLocked()) {
            throw new IllegalArgumentException("Unlock diary first!!");
        }else{
                for(Entry entry : entries) {
                    if(entry.getId() == id) {
                        return entry;
                    }
                }
            throw new IllegalArgumentException("id not found");
        }
    }


    public void delete(int id) {
        if(isLocked){
            System.out.println("Unlock diary first!!");
        }
        Entry toRemove = null;
        for(Entry entry : entries) {
            if(entry.getId() == id) {
                toRemove = entry;

            }
        }
        if(toRemove != null) {
            entries.remove(toRemove);
        }
    }


    public void update(int id,String title, String body) {
        if(isLocked()){
            throw new IllegalArgumentException("Unlock diary first!!");
        }
        Entry toUpdate = findEntry(id);
        if(toUpdate != null) {
            toUpdate.setTitle(title);
            toUpdate.setContent(body);
        }

    }

    public boolean validatePassword(String password) {
        if(password.equals(this.password)) {
            return true;
        }
        return false;
    }
    public String viewEntries() {
        if (entries.isEmpty()) {
            return "No entries available.";
        }

        StringBuilder result = new StringBuilder("Your Entries:\n");
        for (Entry entry : entries) {
            result.append("Entry ID: ").append(entry.getId()).append("\n")
                    .append("Title: ").append(entry.getTitle()).append("\n")
                    .append("Content: ").append(entry.getContent()).append("\n")
                    .append("-------------------\n");
        }
        return result.toString();
    }

}

File: ExtraWork/diaryapp/diary.py
from ExtraWork.diaryapp.entry import Entry
from datetime import datetime

class Diary:
    def __init__(self, username, password):
        self._username = username
        self.password = password
        self._is_locked = True
        self.entries = []

    @property
    def username(self):
        return self._username



    @property
    def is_locked(self):
        return self._is_locked

    def get_id(self):
        return len(self.entries) + 1

    def unlock(self, password):
        if self.validate_password(password):
            self._is_locked = False
        else:
            raise ValueError("Password does not match")

    def lock(self):
        self._is_locked = True

    def create_entry(self, title, body):
        if not self.is_locked:
            entry_id = self.get_id()
            date = datetime.today().date()
            new_entry = Entry(entry_id, title, body,date)
            self.entries.append(new_entry)
            print(f"New entry created\nYour Entry ID: {entry_id}")
        else:
            raise ValueError("Diary is locked, unlock first")

    def find_entry(self, id_number):
        if self.is_locked:
            raise ValueError("Unlock diary first!!")

        for entry in self.entries:
            if entry.id == id_number:
                return entry

        raise ValueError("ID not found")

    def delete_entry(self, id_number):
        if self.is_locked:
            print("Unlock diary first!!")
            return
        to_remove = None
        for entry in self.entries:
            if entry.id == id_number:
                to_remove = entry
                break
        if to_remove:
            self.entries.remove(to_remove)

    def update(self, id_number, title, body):
        if self.is_locked:
            raise ValueError("Unlock diary first!!")
        for entry in self.entries:
            if entry.id == id_number:
                date = datetime.today().date()
                entry = self.find_entry(id)
                entry.title(title)
                entry.content(body)
                entry.date(date)
            else:
                raise ValueError("ID not found")

    def validate_password(self, password):
        return password == self.password



    def view_entries(self):
        output = []
        for entry in self.entries:
            output.append(f"Entry ID: {entry.id}\n"
                          f"Title: {entry.title}\n"
                          f"Content: {entry.content}\n"
                          f"Date: {entry.date}\n"
                          f"-------------------")
        return "\n".join(output)